storage/remote/wal_watcher.go

```diff
@@ -110,6 +110,9 @@ type WALWatcher struct {
 
 	quit chan struct{}
 	done chan struct{}
+
+	// For testing, stop when we hit this segment.
+	maxSegment int
```

Comment:
We should probably initialise this to -1 so it doesn't fail on a fresh Prometheus.